src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -628,22 +620,32 @@ private StudentEnrollDetails enrollStudent(StudentAttributes validStudentAttribu
         return studentList;
     }
 
-    private List<String> getInvalidityInfoInDuplicatedEmail(String email,
-            List<String> studentEmailList, String[] linesArray) {
-        List<String> info = new ArrayList<>();
-        info.add("Same email address as the student in line \"" + linesArray[studentEmailList.indexOf(email) + 1] + "\"");
-        return info;
+    private String getInvalidityInfoInDuplicatedEmail(String[] linesArray, int duplicateEmailIndex) {
+        return "Same email address as the student in line \"" + linesArray[duplicateEmailIndex + 1] + "\"";
     }
 
-    private boolean isStudentEmailDuplicated(String email,
-            List<String> studentEmailList) {
-        return studentEmailList.contains(email);
+    private String addInvalidStudentInfo(StudentAttributes student, String sanitizedLine) {
+        String info = StringHelper.toString(SanitizationHelper.sanitizeForHtml(student.getInvalidityInfo()),
+                "<br>" + Const.StatusMessages.ENROLL_LINES_PROBLEM_DETAIL_PREFIX + " ");
+        return String.format(Const.StatusMessages.ENROLL_LINES_PROBLEM, sanitizedLine, info);
     }
 
-    private boolean isInEnrollList(StudentAttributes student,
-            List<StudentAttributes> studentInfoList) {
+    /**
+     * Returns the index of the first occurrence of the duplicate {@code email} in
+     * {@code studentList}, or -1 if {@code email} is not a duplicate in {@code studentList}.
+     */
+    private int getDuplicateIndex(String email, List<StudentAttributes> studentList) {
+        for (int index = 0; index < studentList.size(); index++) {
+            if (studentList.get(index).email.equalsIgnoreCase(email)) {
+                return index;
+            }
+        }
+        return -1;
+    }
+
+    private boolean isInEnrollList(StudentAttributes student, List<StudentAttributes> studentInfoList) {
         for (StudentAttributes studentInfo : studentInfoList) {
-            if (studentInfo.email.equalsIgnoreCase(student.email)) {
+            if (studentInfo.email.equals(student.email)) {
```

Comment:
This line shouldn't be updated; you've updated the code of the wrong method. Refer to: https://github.com/TEAMMATES/teammates/pull/7930#discussion_r134698402